missed scikit-learn api changes on get_feature_names

### Expected behavior

No calls of `get_feature_names` in favor of `get_feature_names_out`


### Actual behavior

https://github.com/dirty-cat/dirty_cat/blob/fb0a1c4216533034e7516efc0698c7e4477b0243/dirty_cat/super_vectorizer.py#L471


```
 trans_feature_names = trans.get_feature_names(cols) 
pygraphistry/lib/python3.8/site-packages/dirty_cat/super_vectorizer.py:471: in get_feature_names_out
    trans_feature_names = trans.get_feature_names(cols)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
args = (OneHotEncoder(), ['user', 'type']), kwargs = {}
    @functools.wraps(fun)
    def wrapped(*args, **kwargs):
>       warnings.warn(msg, category=FutureWarning)
E       FutureWarning: Function get_feature_names is deprecated; get_feature_names is deprecated in 1.0 and will be removed in 1.2. Please use get_feature_names_out instead.
pygraphistry/lib/python3.8/site-packages/sklearn/utils/deprecation.py:87: FutureWarning
```

See also: https://github.com/dirty-cat/dirty_cat/search?q=get_feature_names

### Steps and code to reproduce bug

https://github.com/graphistry/pygraphistry/runs/6734606058?check_suite_focus=true

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -5,7 +5,7 @@ Bug-fixes
 ---------
 
 * Fixed a bug in the :class:`SuperVectorizer` causing a `FutureWarning`
-  when using the `get_feature_names` method.
+  when using the `get_feature_names_out` method.
 
 Release 0.2.1
 =============

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -1,3 +1,12 @@
+Release 0.2.2
+=============
+
+Bug-fixes
+---------
+
+* Fixed a bug in the :class:`SuperVectorizer` causing a `FutureWarning`
+  when using the `get_feature_names` method.
+
 Release 0.2.1
 =============
 

--- a/dirty_cat/super_vectorizer.py
+++ b/dirty_cat/super_vectorizer.py
@@ -468,7 +468,10 @@ class SuperVectorizer(ColumnTransformer):
             if not hasattr(trans, 'get_feature_names'):
                 all_trans_feature_names.extend(cols)
             else:
-                trans_feature_names = trans.get_feature_names(cols)
+                if _sklearn_loose_version < LooseVersion('1.0'):
+                    trans_feature_names = trans.get_feature_names(cols)
+                else:
+                    trans_feature_names = trans.get_feature_names_out(cols)
                 all_trans_feature_names.extend(trans_feature_names)
 
         if len(ct_feature_names) != len(all_trans_feature_names):

